### PR TITLE
fix: fix stats call for orgs where user is not an approver

### PR DESCRIPTION
**Pull request by bistaastha**

### CODE REVIEW

File: src/app/core/services/tasks.service.ts [hunk 1]
```diff
@@ -476,12 +476,24 @@ export class TasksService {
...
-        })
-      )
+      switchMap((eou) => {
+        if (eou.ou.roles.includes('APPROVER')) {
+          return this.approverReportsService.getReportsStats({
+            next_approver_user_ids: `cs.[${eou.us.id}]`,
+            state: `eq.${ReportState.APPROVER_PENDING}`,
+          });
+        }
+        const zeroResponse: PlatformReportsStatsResponse = {
```
Comment (bistaastha) [inline src/app/core/services/tasks.service.ts:486]: This matches up with the current logic from public APIs, where for non approvers the report count and amount is returned as 0.

File: src/app/core/services/tasks.service.ts [hunk 2]
```diff
@@ -476,12 +476,24 @@ export class TasksService {
...
     return from(this.authService.getEou()).pipe(
-      switchMap((eou) =>
-        this.approverReportsService.getReportsStats({
-          next_approver_user_ids: `cs.[${eou.us.id}]`,
-          state: `eq.${ReportState.APPROVER_PENDING}`,
-        })
-      )
+      switchMap((eou) => {
+        if (eou.ou.roles.includes('APPROVER')) {
+          return this.approverReportsService.getReportsStats({
```
Comment (bistaastha) [inline src/app/core/services/tasks.service.ts:481]: This API call throws a 403 for non approver roles

**Review by mvaishnavi — APPROVED** [review]
please to test well and merge in.

File: src/app/core/services/tasks.service.spec.ts [hunk 3]
```diff
@@ -312,6 +313,23 @@ describe('TasksService', () => {
...
+
+    humanizeCurrencyPipe.transform
+      .withArgs(expectedReportStats.report.total_amount, homeCurrency, true)
+      .and.returnValue('733.48K');
+    humanizeCurrencyPipe.transform
+      .withArgs(expectedReportStats.report.total_amount, homeCurrency)
+      .and.returnValue('₹733.48K');
+
+    tasksService.getTeamReportsTasks().subscribe((teamReportsTasks) => {
+      console.log(teamReportsTasks);
```
Comment (suyashpatil78) [inline src/app/core/services/tasks.service.spec.ts:328]: remove console.log
Also, what are we trying to test here in this spec

**Review by suyashpatil78 — CHANGES_REQUESTED** [review]

**Review by suyashpatil78 — APPROVED** [review]

**Review by Chethan-Fyle — APPROVED** [review]

**Pull request merged**